stuffer/s2n_stuffer_base64.c

```diff
@@ -74,9 +74,12 @@ bool s2n_is_base64_char(unsigned char c)
  */
 int s2n_stuffer_read_base64(struct s2n_stuffer *stuffer, struct s2n_stuffer *out)
 {
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(stuffer));
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(out));
     uint8_t pad[4];
     int bytes_this_round = 3;
-    struct s2n_blob o = {.data = pad,.size = sizeof(pad) };
+    struct s2n_blob o;
```

Comment:
https://github.com/awslabs/s2n/blob/c1cf61787b92884ff9e476e56154f73785415ae7/utils/s2n_blob.h#L49
Use `s2n_stack_blob` for all the stack blobs.

Reply:
I think using `s2n_blob_init` is cleaner here than
```c
    s2n_stack_blob(o, sizeof(pad), sizeof(pad));
    o.data = pad;
```
What do you think? any other suggestion for `maximum`?

Reply:
no, it just would be 

```
s2n_stack_blob(o, 4, 4);
```

The whole point of that macro is to avoid the need to manually declare the array.

Reply:
what about `pad`? you still need to assign it.

Reply:
read the macro for s2n_stack_blob. https://github.com/awslabs/s2n/blob/c1cf61787b92884ff9e476e56154f73785415ae7/utils/s2n_blob.h#L49

Reply:
Oh, yep! Thanks for the tip! Implemented as suggested.